File: benchmarks/ruby/map_lookup.rb
n = (ARGV[0] || "200000").to_i
index = {}
n.times { |i| index["k" + i.to_s] = i }
total = 0
n.times { |i| total = total + index["k" + i.to_s] % 1000 }
puts "#{total} #{index.size}"

File: benchmarks/lua/map_lookup.lua
local n = tonumber(arg[1] or "200000")
local index = {}
local count = 0
for i = 0, n - 1 do
    local key = "k" .. i
    if index[key] == nil then
        count = count + 1
    end
    index[key] = i
end
local total = 0
for i = 0, n - 1 do
    total = total + index["k" .. i] % 1000
end
print(total .. " " .. count)
